thirdparty/faiss/faiss/IndexIVFFlat.h

```diff
@@ -21,6 +21,34 @@ namespace faiss {
  * pre-selects the vectors to be searched, but they are not otherwise
  * encoded, the code array just contains the raw float entries.
  */
+
+struct IVFFlatIteratorWorkspace {
+    IVFFlatIteratorWorkspace(
+            const float* query_data,
+            const IVFSearchParameters* search_params)
+            : query_data(query_data), search_params(search_params) {}
+    ~IVFFlatIteratorWorkspace() {
+        if (scanner) {
```

Comment:
not sure if it's feasible, but can we try to move code in `if (!workspace->initial_search_done) {` that initializes `IVFFlatIteratorWorkspace` members into the constructor? so that `if (!workspace->initial_search_done) {` simply performs the initial search.